src/QMCWaveFunctions/Fermion/MatrixDelayedUpdateCUDA.h

```diff
@@ -369,27 +402,35 @@ class MatrixDelayedUpdateCUDA
   }
 
   template<typename TREAL>
-  inline void mw_invert_transpose(const RefVector<This_t>& engines,
+  static void mw_invert_transpose(const RefVectorWithLeader<This_t>& engines,
                                   const RefVector<const Matrix<T>>& logdetT_list,
                                   const RefVector<std::complex<TREAL>>& LogValues)
   {
+    auto& engine_leader = engines.getLeader();
     // make this class unit tests friendly without the need of setup resources.
-    if (!cuda_handles_)
+    if (!engine_leader.cuda_handles_)
     {
       app_warning() << "MatrixDelayedUpdateCUDA : This message should not be seen in production (performance bug) runs "
                        "but only unit tests (expected)."
                     << std::endl;
-      cuda_handles_ = std::make_unique<CUDALinearAlgebraHandles>();
+      engine_leader.cuda_handles_ = std::make_unique<CUDALinearAlgebraHandles>();
+    }
+    if (!engine_leader.mw_mem_)
```

Comment:
I saw these message indeed in WF optimization runs. So keep them is necessary until we fix unit tests and WF optimization. Since they are not showing up frequently, I just leave the problem as it is now.